### DISCUSSION TITLE
Multi-monitor. One taskbar per screen

### DISCUSSION THREAD

**Discussion by jhoward15** [lawand-dot-io/taskbar#19] [Feature requests]
I'd like to have the option of a different taskbar per screen that only shows the apps on that screen. That's a key feature for me

**Comment by lawand-dot-io** [comment]
Hello. Thanks for your request. This has been requested by several users and is now planned. Not sure when this will be implemented but hopefully it won't take too long. Cheers :)

**Reply by semaj1919** [reply]
Also, it'd be cool if there was an option that allowed for showing all windows OR just the windows on that screen.

**Reply by lawand-dot-io** [reply]
Hi. What do you mean? Basically showing one taskbar per screen and also showing windows of all screens on each of those taskbars?

**Reply by SPavuna** [reply]
He probably means like it is on Windows, you can have a setting that either shows all open apps on all screens, or that each screen shows only the apps on that screen.

**Reply by lawand-dot-io** [reply]
I understand. I can add such an option, sure :)

**Reply by semaj1919** [reply]
> He probably means like it is on Windows, you can have a setting that either shows all open apps on all screens, or that each screen shows only the apps on that screen.

Exactly - a taskbar on every screen and the option for those taskbars to show either: a) only the windows on the screen that the taskbar pertains to -OR- b) the windows on all screens.

**Reply by extide** [reply]
Any ETA on this? I am not willing to spend money on a license unless you have this feature.

**Reply by lawand-dot-io** [reply]
@extide Hello. An ETA is not completely clear, however I will keep Taskbar free until this is implemented.

**Comment by SteveMattan** [comment]
Another vote for having an option to show all apps across all spaces at the same time in the Taskbar. I often use uBar to select an app in another space. Thanks.

**Reply by lawand-dot-io** [reply]
Hmm. That should be easy to implement but I'm trying to understand, it sounds like this is different from showing one taskbar per screen. it sounds like you're saying it's one screen and one taskbar but multiple spaces. or do you mean multiple screens and multiple taskbars and all taskbars show windows from all screens and all spaces?

**Reply by SteveMattan** [reply]
My set up has four monitors and ten spaces. I would want the Taskbar on every monitor and I would want every open app in every space on every task bar. 


Thus no matter which monitor or which space, I could drop to the taskbar click on the desired app, and be taken to the app in the space it is open in. In the case the app is open in multiple spaces, clicking the app on the taskbar should display a list of open windows for selection. 

Hope this makes sense.

**Reply by semaj1919** [reply]
I also understand this request.  

When using just the laptop with multiple spaces - it's a little tedious having to swap between spaces to find the relevant window I'm looking for.  Having the windows for all spaces shown on the taskbar would be useful, because then I could click the relevant window and be moved to that corresponding space.

That said, I'm not really a fan of having the windows of an app condensed into a single label/bucket and then selecting the window I want - it's an extra step.

**Reply by lawand-dot-io** [reply]
@SteveMattan ok, I understand, I can do that. I can allow configuring how many taskbars are shown (only 1 on main display/1 on each display) and which windows they show 👍

**Reply by lawand-dot-io** [reply]
@semaj1919 yeah for sure. my plan ever since I started working on taskbar is to make window switching as easy and simple as possible, so at the moment you see individual items for the windows, and in case I add an option later to condense them into one icon then it will still be an option to show all windows individually :)

**Comment by livinlicious** [comment]
Without wanting to be obnoxious, is there an update on this, or an ETA for this feature?
Thanks for keeping us updated.

**Reply by lawand-dot-io** [reply]
Hello. It's difficult to predict this but probably a couple of months since I will need to do thorough testing to this feature. Cheers

**Reply by SpaceCowboy55** [reply]
Hi - just wondering if you've made any progress on this.  Anxiously awaiting this feature (and multi-row).

**Reply by lawand-dot-io** [reply]
@SpaceCowboy55 hello. Multi-monitor is my top priority and will be in the next version. It might take 1-2 months because I am testing it extensively. Let me know if you wanna be a beta tester :)

**Reply by SpaceCowboy55** [reply]
> @SpaceCowboy55 hello. Multi-monitor is my top priority and will be in the next version. It might take 1-2 months because I am testing it extensively. Let me know if you wanna be a beta tester :)

Yes, please sign me up as a beta tester!

**Reply by lawand-dot-io** [reply]
> > @SpaceCowboy55 hello. Multi-monitor is my top priority and will be in the next version. It might take 1-2 months because I am testing it extensively. Let me know if you wanna be a beta tester :)
> 
> Yes, please sign me up as a beta tester!

cool, send me an email to info@lawand.io

**Reply by pnowy** [reply]
Hi, @lawand-dot-io this is also MUST-HAVE a feature for me (but I would be happy to pay for the tool if that multimonitor support works). If you want I can be a beta tester too :)

**Reply by lawand-dot-io** [reply]
@pnowy cool :) send an email to info@lawand.io

**Comment by swxxii** [comment]
This app is so awesome! would love to get taskbar on all screens though

**Reply by lawand-dot-io** [reply]
working on it 👍

**Comment by e-volusian** [comment]
I agree with @extide -- it's not really usable without this feature, which is a shame because it is the best of all the existing taskbar apps otherwise.

Also, +1 for having the option to show ALL windows on ALL copies of the taskbar

**Reply by lawand-dot-io** [reply]
I agree that this is a crucial feature on a multi-monitor setup 👍

**Comment by lawand-dot-io** [comment]
@e-volusian @swxxii @pnowy @SpaceCowboy55 @livinlicious @extide @semaj1919 @SPavuna @jhoward15 

Good news. This feature is now implemented in version 1.4.2 which is now released. Please report any bugs. Here is some more info:

- Each screen has its own taskbar that only shows that screen’s windows.
- An option is available to make all taskbars show windows from all screens.
- Individual taskbars can be hidden if needed.

**Reply by pnowy** [reply]
Testing a little bit - so far so good and no issues 😄

**Reply by lawand-dot-io** [reply]
@pnowy that's good to hear

**Reply by livinlicious** [reply]
also no issue so far, its been great. 
you created a killer app now, you beat any app on the market. (none work even close as good as taskbar)

**Reply by lawand-dot-io** [reply]
@livinlicious Happy to hear you have no issues after the update. Thanks for your nice comments, you're always a great supporter of the app. Cheers
